content/general/functions.textile

```diff
@@ -8,17 +8,176 @@ Reactor Functions allow your server-less functions to be invoked following chann
+- site := an internal site identifier indicating the data center from which the message was published
+- ruleId := an ID for the rule which has created this request
+- messages := an @Array@ of messages with fields for each message described below
+
+Each member of @messages@ has the following fields:
+
+- id := unique ID assigned by Ably to this message
+- name := a string representing the event name for the published message, see the "publish method":/rest/
+- connectionId := the public unique identifier for the publisher's connection. Find out more about "@connectionId@":/realtime/connection#id
+- timestamp := the time in milliseconds since the epoch when this message was received by Ably
+- data := the data payload for the published message. Find out more about "message data":/rest/channels-messages#message
+
+h4. Decoding messages
+
+Each member of @messages@ is a 'raw' message. Messages sent "over the realtime service":/realtime/channels-messages are automatically decoded into "@Messages@":/realtime/types#message object by the Ably client library. With Reactor Functions you need to to do this explicitly, using "@Message.fromEncodedArray@":/realtime/channels-messages#message-from-encoded-array on the @messages@ array, or "@Message.fromEncoded@":/realtime/channels-messages#message-from-encoded on an individual member of that array. This will transform them into an array of "@Messages@":/realtime/types#message (or in the case of @fromEncoded@, an individual "@Message@":/realtime/types#message). This has several advantages, e.g.:
+
+* It will fully decode any @data@ (using the @encoding@) back into the same datatype that it was sent in (or an equivalent in each client library's language)
+* If you are using "encryption":/realtime/encryption, you can pass your encryption key to the method and it will decrypt the @data@ for you
+
+We recommend you do this for all messages you receive through Reactor Functions. For example (using ably-js):
+
+```[javascript]
+  const messages = Ably.Realtime.Message.fromEncodedArray(request.messages);
+  context.log(messages.toString());
+```
+
+
+h4. Example @channel.message@ JSON payload
+
+```[json]
+{
+   "items":[
+      {
+         "source":"channel.message",
+         "appId":"ABCDef"
+         "channel":"your_channel",
+         "site":"eu-west-2-A",
+         "ruleId":"rUlEID",
+         "messages":[
+            {
+               "id":"9qaOH1C4tO:2:0",
+               "name":"foo",
+               "connectionId":"9qaOH1C4tO",
+               "timestamp":1479301189856,
+               "data":"bar"
+            }
+         ]
+      }
+   ]
+}
+```
+
+Please note that if you are planning to receive messages via Ably Functions, it is theoretically very easy to exceed the "transport rate limits":#transport we impose on Reactor Function to prevent DoS attacks against your own servers. We recommend you consider "message queues":/general/queues instead as they are far more scalable.
```

Comment:
Well message queues are more scalable, but not necessarily far more scalable.  So perhaps include a reference to the REactor Firehose too.

Reply:
We need more specifics here.  We need to link to documentation on limits customers need to be aware of.